guard.cpp

```diff
@@ -18,31 +18,63 @@ void guardList()
     std::cout << "ID       | ERROR    |  Type  | Path " << std::endl;
     for (const auto& elem : records)
     {
-        if ((elem.errType != GARD_Reconfig) &&
-            (elem.recordId != GUARD_RESOLVED))
+        if (displayResolved)
```

Comment:
Please simplify the code both if and else doing the same only condition is different.

Reply:
Please clarify this

Reply:
Already simplified the code till extend possible. Both if and else condition are different.

Reply:
What I think we can do like below
```
void guardList(bool displayResolved)
{
    auto records = getAll();
    if (!records.size())
    {
        std::cout << "No Records to display" << std::endl;
        return;
    }

    bool isHeaderPrinted = false;
    for (const auto& elem : records)
    {
        // Don't print Reconfig (aka Fake record which is used for internal
        // purpose by BMC and Host application)
        if (elem.errType == GARD_Reconfig)
        {
            continue;
        }
        // User interested to see only resolved records
        else if (displayResolved && (elem.recordId != GUARD_RESOLVED))
        {
            continue;
        }
        // User interested to see only unresolved records
        else if (!displayResolved && (elem.recordId == GUARD_RESOLVED))
        {
            continue;
        }

        // Don't print the header if already printed header since
        // records mixed of resolved and unresolved records so if have
        // only either one in retrieved records and user tried to see opposite one
        // then we should not print header else user will get confused. 
        if (!isHeaderPrinted)
        {
            std::cout << "ID       | ERROR    |  Type  | Path " << std::endl;
            isHeaderPrinted = true;
        }

        std::cout << std::hex << std::setw(8) << std::setfill('0')
                  << elem.recordId;

        std::cout << " | ";
        std::cout << std::hex << std::setw(8) << std::setfill('0')
                  << elem.elogId;

        std::cout << " | ";
        std::optional<std::string> gReasonToStr =
                  guardReasonToStr(elem.errType);
        std::cout << *gReasonToStr;

        std::cout << " | ";
        std::optional<std::string> physicalPath =
                  getPhysicalPath(elem.targetId);
        if (!physicalPath)
        {
            std::cout << "Unknown ";
        }
        else
        {
            std::cout << *physicalPath;
        }
        std::cout << std::endl;
    }

    if (!isHeaderPrinted)
    {
        std::cout << "No " << (displayResolved == true ? "resolved" : "unresolved")
                        << " records to display" << std::endl;
    }
}
```

Did you see any issue with this?